iqa_common/executor/execution.py

```diff
@@ -10,15 +13,24 @@
 Defines the representation of a command Execution that is generated
 by the Executor implementations when a command is executed.  
 """
+# Logger
+logger = logging.getLogger(__name__)
 
 
-class Execution(Process):
+class ExecutionException(Exception):
     """
-    The process that has been started by an Executor instance.
+    Exception thrown if a given Execution instance could be created
+    """
+    pass
+
+
+class Execution(abc.ABC):
+    """
+    Represents the execution of a process that has been started by an Executor instance.
     It wraps the command that was triggered as well as the executor
     who generated the Execution instance.
     """
-    def __init__(self, command: Command, executor, modified_args: list=[], env=None):
+    def __init__(self, command: Command, executor, modified_args: list=None, env=None):
```

Comment:
Could you please document those 4 parameters?
I have no idea what env and modified_args is.

We should probably make a rule, to properly document high level API (abstract classes, etc).